### PR TITLE
```is_back_tracking``` docstring update and minor cleanup

### PR DESCRIPTION
**Pull request by eleftherioszisis**

### CODE REVIEW

**Comment by eleftherioszisis** [comment]
Any ideas why locally I get the following test FAIL:

```
======================================================================
FAIL: test_view.test_soma
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/zisis/Projects/NEUROM/NeuroM/neurom_test_venv/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/zisis/Projects/NEUROM/NeuroM/neurom/view/tests/test_view.py", line 74, in test_soma
    nt.ok_(np.allclose(ax.get_ylim(), (0.0, 0.20)) )
AssertionError: None

```

File: neurom/check/morphtree.py [hunk 1]
```diff
@@ -86,12 +87,26 @@ def is_flat(neurite, tol, method='tolerance'):
 
 
 def is_back_tracking(neurite):
-    ''' zigzag
+    ''' Check if a neurite process backtracks to a previous node. As back-tracking is taking place
```
Comment (juanchopanza) [inline neurom/check/morphtree.py]: The sentence starting with "As" reads a bit strangely. Maybe "back-tracking takes place when..."? I think it is enough to say the volumes intersect.

File: neurom/check/morphtree.py [hunk 2]
```diff
@@ -86,12 +87,26 @@ def is_flat(neurite, tol, method='tolerance'):
...
+    when a daughter of a branching process goes back and either overlaps with a previous point, or
+    lies inside the cylindrical volume of the latter. This takes place in the lifetime of a single
```
Comment (juanchopanza) [inline neurom/check/morphtree.py]: "lifetime" here is a bit confusing. Does time have anything to do with it?

**Comment by juanchopanza** [comment]
Could you try to rebase before pushing next time? 6 commits seems like a lot for this change.

**Comment by eleftherioszisis** [comment]
I was afraid you were going to say about the rebase. I will try it tomorrow and hope for the best.

**Comment by juanchopanza** [comment]
You should get used to doing PRs from a branch that is up to date with the remote master. It makes life so much easier.

File: neurom/check/morphtree.py (hunk 1 above)
Comment (eleftherioszisis) [inline neurom/check/morphtree.py]: Shouldn't be there. I removed it. Also removed the detailed description of the algorithm.

File: neurom/check/morphtree.py (hunk 2 above)
Comment (eleftherioszisis) [inline neurom/check/morphtree.py]: Nope. Removed it.

**Comment by eleftherioszisis** [comment]
Can we merge this this time? It will save me a lot of time from fiddling with rebase now that changes have been applied. Next time I will rebase while pulling, before applying changes, to avoid this.

**Pull request merged**